### PR TITLE
DYNModelOmegaRef extension implementation

### PR DESCRIPTION
**Pull request by marcosmc**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Feature

**What is the new behavior (if this is a feature change)?**
Implement the DSL to support the DYNModelOmegaRef dynamic model and the processor of the DSL

### CODE REVIEW

File: src/main/java/com/powsybl/dynawo/xml/DynawoXmlConstants.java [hunk 1]
```diff
@@ -23,6 +23,8 @@
 
     public static final String MACRO_STATIC_REFERENCE_PREFIX = "MSR_";
     public static final String MACRO_CONNECTOR_PREFIX = "MC_";
+    public static final String MACRO_CONNECTOR_TO_GENERATOR_SUFIX = "ToGenerator";
```
Comment (zamarrenolm) [inline src/main/java/com/powsybl/dynawo/xml/DynawoXmlConstants.java:26]: I would move these two constants to the `DYNModelOmegaRef` class, as they are specific of the connectors related to it.

Comment (zamarrenolm) [reply src/main/java/com/powsybl/dynawo/xml/DynawoXmlConstants.java:26]: I was referring to `MACRO_CONNECTOR_TO_GENERATOR_SUFIX` and `MACRO_CONNECTOR_TO_NUMCCMACHINE_SUFIX`. By the way, there is a typo, it should be spelled `SUFFIX`.

Comment (marcosmc) [reply src/main/java/com/powsybl/dynawo/xml/DynawoXmlConstants.java]: Done

File: src/main/groovy/com/powsybl/dynawo/simulator/DYNModelOmegaRefGroovyExtension.groovy [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    void load(Binding binding, Consumer<DynamicModel> consumer) {
+        binding.OmegaRef = { Closure<Void> closure ->
+            def cloned = closure.clone()
+            DYNModelOmegaRefSpec dynModelOmegaRefSpec = new DYNModelOmegaRefSpec()
+
+            cloned.delegate = dynModelOmegaRefSpec
+            cloned()
+
+            if (!dynModelOmegaRefSpec.generatorDynamicModelId) {
+                throw new DslException("'generatorModelId' field is not set")
```
Comment (agnesLeroy) [inline src/main/groovy/com/powsybl/dynawo/simulator/DYNModelOmegaRefGroovyExtension.groovy:47]: generatorDynamicModelId

File: src/main/java/com/powsybl/dynawo/dyd/DYNModelOmegaRef.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    private static final String OMEGA_REF_PARAMETER_SET_ID = "OMEGA_REF";
+    private static final String MACRO_CONNECTOR_TO_GENERATOR_SUFFIX = "ToGenerator";
+    private static final String MACRO_CONNECTOR_TO_NUMCCMACHINE_SUFFIX = "ToNumCCMachine";
+
+    // OmegaRef is a magic model
+    // There will be multiple instances of OmegaRef dynamic model,
+    // each one having only one reference to a generator dynamic model
+    // When all these OmegaRef objects are serialized to XML,
+    // Only one blackBoxModel must be present in the XML output,
+    // connected to all generator dynamic models that have been specified
```
Comment (agnesLeroy) [inline src/main/java/com/powsybl/dynawo/dyd/DYNModelOmegaRef.java:37]: Replace this comment by:
// OmegaRef is a special dynamic model
// Its role is to synchronize the generators' frequency
// There will be multiple Java instances of the OmegaRef dynamic model, one for
// each generator's dynamic model connected to it.
// The corresponding black box model XML entry is serialized only once.
// For each generator synchronised through the OmegaRef model, 
// there will be one XML entry for the connection with the generator's dynamic model,
// and one XML entry for the connection with the NETWORK dynamic model.
// There are thus two macroConnectors defined for OmegaRef: one to connect it to a
// generator's dynamic model and one to connect it to the NETWORK model.

File: src/main/java/com/powsybl/dynawo/dyd/DYNModelOmegaRef.java [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+            // Special magic here:
+            // All instances of DYNModelOmegaRef refer in fact to a single Dynawo BlackBoxModel
+            // We write the blackBoxModel in the output XML only once
+            writer.writeEmptyElement(DYN_URI, "blackBoxModel");
+            writer.writeAttribute("id", getDynamicModelId());
+            writer.writeAttribute("lib", getLib());
+            writer.writeAttribute("parFile", OMEGAREF_PAR_FILENAME);
+            writer.writeAttribute("parId", getParameterSetId());
+        }
+        // This instance of DYNModelOmegaRef has a reference to one generator, write its connect
```
Comment (agnesLeroy) [inline src/main/java/com/powsybl/dynawo/dyd/DYNModelOmegaRef.java:74]: This instance of DYNModelOmegaRef has a reference to one generator, write its connect and the subsequent connect to the NETWORK model

File: src/main/groovy/com/powsybl/dynawo/simulator/DYNModelOmegaRefGroovyExtension.groovy (hunk 2 above)
Comment (marcosmc) [reply src/main/groovy/com/powsybl/dynawo/simulator/DYNModelOmegaRefGroovyExtension.groovy]: Done

File: src/main/java/com/powsybl/dynawo/dyd/DYNModelOmegaRef.java (hunk 3 above)
Comment (marcosmc) [reply src/main/java/com/powsybl/dynawo/dyd/DYNModelOmegaRef.java]: Done

File: src/main/java/com/powsybl/dynawo/dyd/DYNModelOmegaRef.java (hunk 4 above)
Comment (marcosmc) [reply src/main/java/com/powsybl/dynawo/dyd/DYNModelOmegaRef.java]: Done

**Pull request merged**